### PR TITLE
ci: new release flow

### PR DESCRIPTION
**Pull request by ypoplavs**
## Pull request description 
**Changes in workflow:**

-   `docker-build-develop.yaml` is triggered on every push to `develop` branch. It builds API and executors with commit hash tag and starts deployment to `develop` cluster from `helm-charts` repo, passing the tag as a client payload. API is always built, executors - ONLY on directory change!
- `docker-build-release.yaml` is triggered on every push to `release` branch. It builds API and executors with commit hash tag and starts deployment to `stage` cluster from `helm-charts` repo. Creates a release branch in `helm-charts` repo if it doesn't exist. The same as in `develop` flow - API is always built, executors - ONLY on directory change!
- `docker-build-api-executors-beta-tag.yaml` is triggered on creation of Beta releases for API and executors and deployed to pre-release k8s cluster.
- `docker-build-api-executors-tag.yaml` is triggered on tag creation. It builds API and executors with the necessary image tags, updates Chart.yaml files in `helm-charts` repo.

**Changes in Dockerfiles**

Refactored Dockerfiles so that they do not build the app, but just copy the binary, which was built by GoReleaser. This eliminated the need for `bake` files, which were removed. 

**Other changes**
Renamed `executor-init` and `executor-scraper` to `init-executor` and `scraper-executor`. Respective Docker registries were created. 
 
## Checklist (choose whats happened)

- [x] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes
Reformated Dockerfiles of executors.
-

## Changes

-

## Fixes

-

### CODE REVIEW

File: .github/workflows/docker-build-api-executors-tag.yaml [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+        distribution: goreleaser
+        version: latest
+        args: release -f goreleaser_files/.goreleaser-docker-build-executor-playwright.yml
+      env:
+        GITHUB_TOKEN: ${{ secrets.CI_BOT_TOKEN }}
+        DOCKER_BUILDX_BUILDER: "${{ steps.buildx.outputs.name }}"
+        DOCKER_BUILDX_CACHE_FROM: "type=gha"
+        DOCKER_BUILDX_CACHE_TO: "type=gha,mode=max"
+
+#  update_api_helm_charts:
```
Comment (vsukhin) [inline .github/workflows/docker-build-api-executors-tag.yaml:441]: better to remove or uncomment if it's used

File: .github/workflows/docker-build-develop.yaml [hunk 2]
```diff
@@ -0,0 +1,566 @@
+name: Docker images build develop
+on:
+  push:
+    branches:
+      - develop
+    #  - switch-to-goreleaser
```
Comment (vsukhin) [inline .github/workflows/docker-build-develop.yaml:6]: remove comment in ready pr

File: .github/workflows/docker-build-release.yaml [hunk 3]
```diff
@@ -0,0 +1,593 @@
+name: Docker images build release
+on:
+  push:
+    branches:
+      - release_**
+      - switch-to-goreleaser
```
Comment (vsukhin) [inline .github/workflows/docker-build-release.yaml:6]: is this for testing?

File: build/api-server/Dockerfile [hunk 4]
```diff
@@ -1,38 +1,8 @@
...
-    GOOS=$TARGETOS GOARCH=$TARGETARCH  go build \
-        -ldflags "-X github.com/kubeshop/testkube/pkg/version.Version=$VERSION \
-            -X github.com/kubeshop/testkube/pkg/version.Commit=$COMMIT \
-	    -X github.com/kubeshop/testkube/internal/app/api/v1.SlackBotClientID=$SLACK_BOT_CLIENT_ID \
-	    -X github.com/kubeshop/testkube/internal/app/api/v1.SlackBotClientSecret=$SLACK_BOT_CLIENT_SECRET \
-            -X github.com/kubeshop/testkube/pkg/telemetry.TestkubeMeasurementID=$ANALYTICS_TRACKING_ID \
-            -X github.com/kubeshop/testkube/pkg/telemetry.TestkubeMeasurementSecret=$ANALYTICS_API_KEY \
-            -X github.com/kubeshop/testkube/pkg/telemetry.SegmentioKey=$SEGMENTIO_KEY \
-            -X github.com/kubeshop/testkube/pkg/telemetry.CloudSegmentioKey=$CLOUD_SEGMENTIO_KEY" \
-        -o /app -mod mod -a .
```
Comment (vsukhin) [inline build/api-server/Dockerfile:28]: did we move these flags to go releaser?

**Review by vsukhin — COMMENTED** [review]
looks in general, I don't see anything weird

File: .github/workflows/docker-build-develop.yaml [hunk 5]
```diff
@@ -0,0 +1,566 @@
...
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Step
```
Comment (dejanzele) [inline .github/workflows/docker-build-develop.yaml:46]: "Step" doesn't seem as an intuitive name

**Pull request merged**